### PR TITLE
Channel helper

### PR DESCRIPTION
**Pull request by movestill**
Fix problem with users not specifying the correct channel data type before using the volume service. See commit messages for details.

### CODE REVIEW

File: intern/remote/boss/remote.py [hunk 1]
```diff
@@ -683,6 +683,21 @@ def list_coordinate_frames(self):
         cf = CoordinateFrameResource(name='')
         return self._list_resource(cf)
 
+    def get_channel(self, chan_name, coll_name, exp_name):
```
Comment (dkleissa) [inline intern/remote/boss/remote.py]: Maybe for this helper the args should be in the logical order (collection, experiment, channel)? Is this useful or more confusing?

File: intern/resource/boss/resource.py [hunk 2]
```diff
@@ -442,6 +459,15 @@ def __init__(self, name, collection_name, experiment_name, type="image",
         self.exp_name = experiment_name
 
         self._type = self.validate_type(type)
+
+        # Class is considered fully initialized if datatype set during
+        # initialization.
+        if datatype == '':
```
Comment (dkleissa) [inline intern/resource/boss/resource.py]: Seems like this is a little misleading. if `datatype` is set, that doesn't mean it necessarily is "fully initialized". Probably at a minimum should make sure everything that is needed for serialization is there?

File: intern/remote/boss/remote.py (hunk 1 above)
Comment (movestill) [inline intern/remote/boss/remote.py]: I thought about this, but thought it'd be better to be consistent with the order used to instantiate resources.

File: intern/resource/boss/resource.py (hunk 2 above)
Comment (movestill) [inline intern/resource/boss/resource.py]: I was looking at this as meaning that the bare minimum is specified so the ChannelResource could be used for a cutout. Maybe the property should be renamed to `channel_cutout_ready` so the meaning is clearer?

**Comment by sandyhider** [comment]
Is this ready to be merged?  It looks like Dean approved it but it was never merged.

**Pull request merged**